pkg/configfilearg/defaultparser_test.go

```diff
@@ -0,0 +1,61 @@
+package configfilearg
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_UnitMustParse(t *testing.T) {
+	tests := []struct {
+		name   string
+		args   []string
+		config string
+		want   []string
+	}{
+		{
+			name: "Basic server",
+			args: []string{"k3s", "server"},
+
+			want: []string{"k3s", "server"},
+		},
+		{
+			name: "Server with known flags",
+			args: []string{"k3s", "server", "-t 12345", "--write-kubeconfig-mode 644"},
+
+			want: []string{"k3s", "server", "-t 12345", "--write-kubeconfig-mode 644"},
+		},
+		{
+			name:   "Server with known flags and config with known and unknown flags",
+			args:   []string{"k3s", "server", "--write-kubeconfig-mode 644"},
+			config: "./testdata/defaultdata.yaml",
+			want: []string{"k3s", "server", "--token=12345", "--node-label=DEAFBEEF",
+				"--etcd-s3=true", "--etcd-s3-bucket=my-backup", "--write-kubeconfig-mode 644"},
+		},
+		{
+			name: "Basic etcd-snapshot",
+			args: []string{"k3s", "etcd-snapshot", "save"},
+
+			want: []string{"k3s", "etcd-snapshot", "save"},
+		},
+		{
+			name: "Etcd-snapshot with known flags",
+			args: []string{"k3s", "etcd-snapshot", "save", "--s3=true"},
+
+			want: []string{"k3s", "etcd-snapshot", "save", "--s3=true"},
+		},
+		{
+			name:   "Etcd-snapshot with config with known and unknown flags",
+			args:   []string{"k3s", "etcd-snapshot", "save"},
+			config: "./testdata/defaultdata.yaml",
+			want:   []string{"k3s", "etcd-snapshot", "save", "--etcd-s3=true", "--etcd-s3-bucket=my-backup"},
+		},
```

Comment:
Should we test loading agent config as well?

Reply:
Added agent subtests